lib/Dialect/FIRRTL/Transforms/LowerXMR.cpp

```diff
@@ -418,6 +420,40 @@ class LowerXMRPass : public LowerXMRBase<LowerXMRPass> {
     return success();
   }
 
+  LogicalResult handleTopModuleRefPorts() {
+    auto topModule = dyn_cast<FModuleOp>(*getOperation().getMainModule());
+    if (!topModule)
+      return success();
+
+    auto builder = ImplicitLocOpBuilder::atBlockBegin(
+        topModule.getLoc(), getOperation().getBodyBlock());
+
+    for (size_t portIndex = 0, numPorts = topModule.getNumPorts();
+         portIndex != numPorts; ++portIndex) {
+      auto refType = topModule.getPortType(portIndex).dyn_cast<RefType>();
+      if (!refType || isZeroWidth(refType.getType()) ||
+          topModule.getPortDirection(portIndex) != Direction::Out)
+        continue;
```

Comment:
We may have to think about how to handle the zero width case. We may need to generate the macro as `'0`. TBD.

Reply:
commented out macro?  What would it point to?  Since 0-width elements are generally removed, I don't see why a ref to a zero width element wouldn't also.